frontend/src/components/CourseDescriptionPanel/CourseAttributes/CourseAttributes.tsx

```diff
@@ -31,6 +38,16 @@ const CourseAttributes = ({ course, courseCapacity }: CourseAttributesProps) =>
 
   const { study_level: studyLevel, terms, campus, code, school, UOC } = course;
 
+  const currentTerm = getMostRecentPastTerm(CURR_YEAR).T;
+
+  const recentTermNumber = parseInt(terms[0].slice(1), 10);
```

Comment:
Terms might be an empty array if there are no offerings of that course, which will cause a out of range error. 

Also are you meant to only be grabbing the 0th element here? For COMP2521, terms is [t0, t1, t2, t3], and for COMP1511, terms is [t1, t2, t3]. Is it just meant to be the earliest available term of this year?

Reply:
uhm i did hte check for the out of range error

but right now i'm comparing 0th element which is the most current term and comparing with the term i fetched using the function getMostRecentPastTerm and comparing it if the term i fetched it larger I would use their one but i'm not sure why its not jumping onto t2, but it is fetching the first term so it can compare